pkg/pillar/utils/netutils/hostnameorip.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2025 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package netutils
+
+import (
+	"encoding/json"
+	"net"
+)
+
+// HostnameOrIP holds either a literal IP address or a hostname.
+// It stores both the original string and the parsed IP (if valid).
+type HostnameOrIP struct {
+	raw string // original string
+	ip  net.IP // parsed IP if raw is a literal IP, nil if hostname
+}
+
+// NewHostnameOrIP creates a HostnameOrIP from a string.
+func NewHostnameOrIP(s string) HostnameOrIP {
+	return HostnameOrIP{
+		raw: s,
+		ip:  net.ParseIP(s),
```

Comment:
Do the callers all check for this being nil?

Reply:
There is a method for that:
```
// IsIP returns true if this is a literal IP.
func (h HostnameOrIP) IsIP() bool {
	return h.ip != nil
}
```